core/common/src/main/java/alluxio/collections/ConcurrentHashSet.java

```diff
@@ -104,6 +104,9 @@ public boolean equals(Object o) {
       return true;
     }
     // allow equality with any Set.
+    if (!(o instanceof ConcurrentHashSet)) {
```

Comment:
@aaudiber I think you are right. @NAP-GHJ , how about instead of modifying the `equals` method, submit a patch to remove this method from this file?